File: CsLibrary/CsLibraryTest/ByteTest.cs
﻿using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace JavaClasses
{
	[TestClass]
	public class ByteTest
	{
        /**
         * Tests the byteValue method.
         */
        [TestMethod]
		public void byteValueTest()
        {
            Byte a;
            sbyte b;
            // minimal value
            a = new Byte(Byte.MIN_VALUE);
            b = a.byteValue();
            Assert.AreEqual(-128, b);
            // zero value
            a = new Byte((sbyte)0);
            b = a.byteValue();
            Assert.AreEqual(0, b);
            // maximal value
            a = new Byte(Byte.MAX_VALUE);
            b = a.byteValue();
            Assert.AreEqual(127, b);
        }

        /**
         * Tests the compareTo method.
         */
        [TestMethod]
		public void compareToTest()
        {
            Byte a, b;
            int c;
            // minimal and bigger negative value
            a = new Byte(Byte.MIN_VALUE);
            b = new Byte((sbyte)-1);
            c = a.compareTo(b);
            Assert.AreEqual(-127, c);
            // maximal and smaller positive value
            a = new Byte(Byte.MAX_VALUE);
            b = new Byte((sbyte)1);
            c = a.compareTo(b);
            Assert.AreEqual(126, c);
            // zero and positive value
            a = new Byte((sbyte)0);
            b = new Byte((sbyte)1);
            c = a.compareTo(b);
            Assert.AreEqual(-1, c);
            // zero and negative value
            a = new Byte((sbyte)0);
            b = new Byte((sbyte)-1);
            c = a.compareTo(b);
            Assert.AreEqual(1, c);
            // both values are positive
            a = new Byte((sbyte)1);
            b = new Byte((sbyte)1);
            c = a.compareTo(b);
            Assert.AreEqual(0, c);
            // both values are negative
            a = new Byte((sbyte)-1);
            b = new Byte((sbyte)-1);
            c = a.compareTo(b);
            Assert.AreEqual(0, c);
        }

        /**
         * Tests the compare method.
         */
        [TestMethod]
		public void compareTest()
        {
            sbyte a, b;
            int c;
            a = Byte.MIN_VALUE;
            b = (sbyte)-1;
            c = Byte.compare(a, b);
            Assert.AreEqual(-127, c);
            a = Byte.MAX_VALUE;
            b = (sbyte)1;
            c = Byte.compare(a, b);
            Assert.AreEqual(126, c);
            a = (sbyte)0;
            b = (sbyte)1;
            c = Byte.compare(a, b);
            Assert.AreEqual(-1, c);
            a = (sbyte)0;
            b = (sbyte)-1;
            c = Byte.compare(a, b);
            Assert.AreEqual(1, c);
            a = (sbyte)1;
            b = (sbyte)1;
            c = Byte.compare(a, b);
            Assert.AreEqual(0, c);
            a = (sbyte)-1;
            b = (sbyte)-1;
            c = Byte.compare(a, b);
            Assert.AreEqual(0, c);
        }

        /**
         * Tests the equals method.
         */
        [TestMethod]
		public void equalsTest()
        {
            Byte a, b;
            bool c;
            a = new Byte(Byte.MIN_VALUE);
            b = new Byte((sbyte)-1);
            c = a.equals(b);
            Assert.AreEqual(false, c);
            a = new Byte(Byte.MAX_VALUE);
            b = new Byte((sbyte)1);
            c = a.equals(b);
            Assert.AreEqual(false, c);
            a = new Byte((sbyte)0);
            b = new Byte((sbyte)1);
            c = a.equals(b);
            Assert.AreEqual(false, c);
            a = new Byte((sbyte)0);
            b = new Byte((sbyte)-1);
            c = a.equals(b);
            Assert.AreEqual(false, c);
            a = new Byte((sbyte)1);
            b = new Byte((sbyte)1);
            c = a.equals(b);
            Assert.AreEqual(true, c);
            a = new Byte((sbyte)-1);
            b = new Byte((sbyte)-1);
            c = a.equals(b);
            Assert.AreEqual(true, c);
        }

        /**
         * Tests the toString method.
         */
        [TestMethod]
		public void toStringTest()
        {
            Byte a;
            String b;
            // positive value text representation
            a = new Byte((sbyte)100);
            b = a.toString();
            Assert.AreEqual(100, b);
            // zero value text representation
            a = new Byte((sbyte)0);
            b = a.toString();
            Assert.AreEqual(0, b);
            // negative value text representation
            a = new Byte((sbyte)-100);
            b = a.toString();
            Assert.AreEqual(-100, b);
        }

        /**
         * Tests the static toString method.
         */
        [TestMethod]
		public void toStringTest2()
        {
            Byte a;
            String b;
            a = new Byte((sbyte)100);
            b = Byte.toString(a);
            Assert.AreEqual(100, b);
            a = new Byte((sbyte)0);
            b = Byte.toString(a);
            Assert.AreEqual(0, b);
            a = new Byte((sbyte)-100);
            b = Byte.toString(a);
            Assert.AreEqual(-100, b);
        }

        /**
         * Tests the parseByte method.
         */
        [TestMethod]
		public void parseByteTest()
        {
            Byte a;
            sbyte b;
            // positive value parsing
            a = Byte.parseByte("100");
            b = a.byteValue();
            Assert.AreEqual(100, b);
            // zero value parsing
            a = Byte.parseByte("0");
            b = a.byteValue();
            Assert.AreEqual(0, b);
            // negative value parsing
            a = Byte.parseByte("-100");
            b = a.byteValue();
            Assert.AreEqual(-100, b);
        }

    }
}

File: C/CLibrary/tests/ByteTest.c
#include <stdio.h>
#include <stdlib.h>

#include "test.h"
#include "Byte.h"

/*
 * Simple C Test Suite
 */

/**
 * Tests the byteValue function.
 */

void byteValueTest() {
    Byte *a;
    int8_t b;
    // minimal value
    a = new_Byte(Byte_MIN_VALUE);
    b = byteValue(a);
    assertEqualsB((int8_t)-128, b);
    // zero value
    a = new_Byte((int8_t) 0);
    b = byteValue(a);
    assertEqualsB((int8_t)0, b);
    // maximal value
    a = new_Byte(Byte_MAX_VALUE);
    b = byteValue(a);
    assertEqualsB((int8_t)127, b);
}

/**
 * Tests the compareTo function.
 */

void compareToTest() {
    Byte *a, *b;
    int32_t c;
    // minimal and bigger negative value
    a = new_Byte(Byte_MIN_VALUE);
    b = new_Byte((int8_t) - 1);
    c = compareToB(a, b);
    assertEqualsI(-127, c);
    // maximal and smaller positive value
    a = new_Byte(Byte_MAX_VALUE);
    b = new_Byte((int8_t) 1);
    c = compareToB(a, b);
    assertEqualsI(126, c);
    // zero and positive value
    a = new_Byte((int8_t) 0);
    b = new_Byte((int8_t) 1);
    c = compareToB(a, b);
    assertEqualsI(-1, c);
    // zero and negative value
    a = new_Byte((int8_t) 0);
    b = new_Byte((int8_t) - 1);
    c = compareToB(a, b);
    assertEqualsI(1, c);
    // both values are positive
    a = new_Byte((int8_t) 1);
    b = new_Byte((int8_t) 1);
    c = compareToB(a, b);
    assertEqualsI(0, c);
    // both values are negative
    a = new_Byte((int8_t) - 1);
    b = new_Byte((int8_t) - 1);
    c = compareToB(a, b);
    assertEqualsI(0, c);
}

/**
 * Tests the compare function.
 */

void compareTest() {
    int8_t a, b;
    int32_t c;
    a = Byte_MIN_VALUE;
    b = (int8_t) - 1;
    c = Byte_compare(a, b);
    assertEqualsI(-127, c);
    a = Byte_MAX_VALUE;
    b = (int8_t) 1;
    c = Byte_compare(a, b);
    assertEqualsI(126, c);
    a = (int8_t) 0;
    b = (int8_t) 1;
    c = Byte_compare(a, b);
    assertEqualsI(-1, c);
    a = (int8_t) 0;
    b = (int8_t) - 1;
    c = Byte_compare(a, b);
    assertEqualsI(1, c);
    a = (int8_t) 1;
    b = (int8_t) 1;
    c = Byte_compare(a, b);
    assertEqualsI(0, c);
    a = (int8_t) - 1;
    b = (int8_t) - 1;
    c = Byte_compare(a, b);
    assertEqualsI(0, c);
}

/**
 * Tests the equals function.
 */

void equalsTest() {
    Byte *a, *b;
    bool c;
    a = new_Byte(Byte_MIN_VALUE);
    b = new_Byte((int8_t) - 1);
    c = equalsB(a, b);
    assertEqualsBool(false, c);
    a = new_Byte(Byte_MAX_VALUE);
    b = new_Byte((int8_t) 1);
    c = equalsB(a, b);
    assertEqualsBool(false, c);
    a = new_Byte((int8_t) 0);
    b = new_Byte((int8_t) 1);
    c = equalsB(a, b);
    assertEqualsBool(false, c);
    a = new_Byte((int8_t) 0);
    b = new_Byte((int8_t) - 1);
    c = equalsB(a, b);
    assertEqualsBool(false, c);
    a = new_Byte((int8_t) 1);
    b = new_Byte((int8_t) 1);
    c = equalsB(a, b);
    assertEqualsBool(true, c);
    a = new_Byte((int8_t) - 1);
    b = new_Byte((int8_t) - 1);
    c = equalsB(a, b);
    assertEqualsBool(true, c);
}

/**
 * Tests the toString function.
 */

void toStringTest() {
    Byte *a;
    String *b;
    // positive value text representation
    a = new_Byte((int8_t) 100);
    b = toStringB(a);
    assertEqualsStr("100", b->s);
    // zero value text representation
    a = new_Byte((int8_t) 0);
    b = toStringB(a);
    assertEqualsStr("0", b->s);
    // negative value text representation
    a = new_Byte((int8_t) - 100);
    b = toStringB(a);
    assertEqualsStr("-100", b->s);
}

/**
 * Tests the static toString function.
 */

void toStringTest2() {
    int8_t a;
    String *b;
    a = (int8_t) 100;
    b = Byte_toString(a);
    assertEqualsStr("100", b->s);
    a = (int8_t) 0;
    b = Byte_toString(a);
    assertEqualsStr("0", b->s);
    a = (int8_t) - 100;
    b = Byte_toString(a);
    assertEqualsStr("-100", b->s);
}

/**
 * Tests the parseByte function.
 */

void parseByteTest() {
    int8_t a;
    // positive value parsing
    a = Byte_parseByte(new_String("100"));
    assertEqualsB((int8_t)100, a);
    // zero value parsing
    a = Byte_parseByte(new_String("0"));
    assertEqualsB((int8_t)0, a);
    // negative value parsing
    a = Byte_parseByte(new_String("-100"));
    assertEqualsB((int8_t)-100, a);
}

int main(int argc, char** argv) {
    const int testCount = 7;
    setUpTestModule("ByteTest", testCount);

    registerTest(byteValueTest, "byteValueTest");
    registerTest(compareToTest, "compareToTest");
    registerTest(compareTest, "compareTest");
    registerTest(equalsTest, "equalsTest");
    registerTest(toStringTest, "toStringTest");
    registerTest(toStringTest2, "toStringTest2");
    registerTest(parseByteTest, "parseByteTest");
    
    runTests();

    return (EXIT_SUCCESS);
}
